ui/components/DataTable.tsx

```diff
@@ -141,6 +173,98 @@ function UnstyledDataTable({
           </TableBody>
         </Table>
       </TableContainer>
+      {/* pagination row */}
+      <Spacer padding="xs" />
+      {!disablePagination && (
```

Comment:
This is complex enough to warrant putting all of this block in it's own component. `<PaginationControls />` or something like that